### PR TITLE
Declare HPOS incompatibility

### PR DESCRIPTION
**Pull request by jessy-p**
#### Changes proposed in this Pull Request

Declare incompatibility of WooCommerce Payments with HPOS: pcShBQ-oY-p2

#### Testing instructions
- Get latest HPOS WooCommerce release from [here](https://github.com/woocommerce/woocommerce/releases/tag/feature-custom-order-table)
- Run the following command using wp cli 
```
wp eval 'do_action("before_woocommerce_init"); print_r(Automattic\WooCommerce\Utilities\FeaturesUtil::get_compatible_plugins_for_feature("custom_order_tables"));'
```
- Should get output similar to below, declaring WooCommerce Payments as an incompatible plugin
```
Array
(
    [compatible] => Array
        (
        )

    [incompatible] => Array
        (
            [0] => woocommerce-payments/woocommerce-payments.php
        )

)
```
- Backward Compatibility: On  7.0/ older versions of WooCommerce, the above command will not run, but the plugin should function without any existing functionality breaking.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] N/A - Covered with tests (or have a good reason not to test in description ☝️)
- [x] N/A - Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) :  'QA Testing Not Applicable'
- [x] N/A - Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by jbordonado** [comment]
WCPay is now a managed plugin on Atomic, and this discussion caught my eye: p1665514564244559-slack-C9QDV4TL4

What does `__FILE__` return in our case? Can we test this PR on an Atomic site?

File: woocommerce-payments.php [hunk 1]
```diff
@@ -323,3 +323,17 @@ function wcpay_tasks_init() {
...
+/**
+ * As the class is defined in later versions of WC, Psalm infers error.
+ *
+ * @psalm-suppress UndefinedClass
+ */
+add_action(
+	'before_woocommerce_init',
+	function() {
+		if ( class_exists( '\Automattic\WooCommerce\Utilities\FeaturesUtil' ) ) {
+			\Automattic\WooCommerce\Utilities\FeaturesUtil::declare_compatibility( 'custom_order_tables', __FILE__, false );
```
Comment (deepakpathania) [inline woocommerce-payments.php]: We can hardcode the plugin name to `woocommerce-payments/woocommerce-payments.php` based on this discussion for atomic sites: p1665562457707829/1665514564.244559-slack-C9QDV4TL4

**Review by deepakpathania — CHANGES_REQUESTED** [review]
Seeing this notice at the top of my site with this change:

```
Notice: Function CustomOrdersTableController::show_feature was called incorrectly.

CustomOrdersTableController::show_feature: The visibility of the custom orders table feature is now handled by the WooCommerce features engine. See the FeaturesController class, or go to WooCommerce - Settings - Advanced - Features. 

Backtrace: require_once('wp-admin/admin.php'), require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, enable_cot, Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController->show_feature, wc_doing_it_wrong Please see Debugging in WordPress for more information. (This message was added in version 7.0.) in /var/www/html/wp-includes/functions.php on line 5831 Call Stack: 0.0002 405880 1. {main}()
```

Left one comment other than that.

**Comment by deepakpathania** [comment]
The notice was due to the COT enabling script, disabling the snippet removed the same.

File: woocommerce-payments.php (hunk 1 above)
Comment (jessy-p) [reply woocommerce-payments.php]: Done

**Review by deepakpathania — APPROVED** [review]
Changes LGTM now based on the P2, haven't tested on an atomic site but since we have hardcoded the plugin name, should be ok.

**Pull request merged**